lib/tictactoe.rb

```diff
@@ -15,22 +15,32 @@ def initialize(presenter, display, board, player)
 
   def run
     show_board
-    until board.game_over?
-      play_turn
-      show_board
-    end
+    play_game
+    show_outcome
   end
 
   private
 
   def show_board
-    display.output(presenter.display_board(board))
+    display.output(presenter.board(board))
   end
 
   def play_turn
     position = player.selection(board)
     board.place_token(position)
   end
+
+  def play_game
+    until board.outcome
+      play_turn
+      show_board
+    end
+  end
+
+  def show_outcome
+    outcome = board.outcome
+    outcome == true ? display.output(presenter.draw_outcome) : display.output(presenter.win_outcome(outcome))
+  end
```

Comment:
This is similar to a scenario that we discussed the other day, except that instead of explicit queries to the board object, the values that are returned have implicit meaning.

i.e.
```ruby
if board.draw?
   presenter.show_draw()
else
   presenter.show_winner(board)
end
```

I think that part of the reason for this is that `#outcome` has two different return types. It will return a boolean, or some value to represent the winner. Ideally, it would only return one type of value.

```ruby
presenter.show_outcome(outcome)
```

Reply:
Good point.  I'll adjust.